### DISCUSSION TITLE
Fail to compile cfmesh on blueCFD-Core-2017

### DISCUSSION THREAD

**Discussion by LeeLeong** [blueCFD/Core#227] [Show and tell]
Hi Bruno Santos,

I try to compile cfmesh on blueCFD-Core-2017 5.x-478c685b424c using link from https://github.com/blueCFD/Core/wiki/Building-cfMesh-with-blueCFD-Core#building-with-bluecfd-core-2017 .
I run into problem step 6th.  Please find the log file in the attachment folder.
[log.make.gz](https://github.com/blueCFD/Core/files/2504767/log.make.gz)

Thank you in advance.

**Comment by wyldckat** [comment]
Hi @LeeLeong,

I've confirmed that the version you are using is blueCFD-Core 2017-2, many thanks for the complete version hash!

Because the reason why it failed to compile is because it is using the full path to the `cfMesh` folder, namely `C:/Program Files/blueCFD-Core-2017/msys64/home/ofuser/cfMesh`, which in the terminal it should be the path `~/cfMesh`.

But if the instructions from https://github.com/blueCFD/Core/wiki/Building-cfMesh-with-blueCFD-Core#building-with-bluecfd-core-2017 - are strictly followed, then this issue should not have occurred, given the instructions aim to place the `cfMesh` folder at the `~/blueCFD/ofuser-of5`, namely `C:\Program Files\blueCFD-Core-2017\ofuser-of5`.

So my guess is that you've downloaded manually the cfMesh package instead of using the git command?

----

**Possibility A**: If this was the reason why you've placed it the `ofuser` home folder, then my request is that you please try the following steps:

1. Open a new the blueCFD-Core terminal.

2. Run the following commands:

    ```
    mkdir -p $FOAM_RUN
    user
    explorer .
    ```

    Note: There is a space character and then a dot symbol at the end of the 3rd command, just in case it's not visible.

3. Windows Explorer should have opened onto the folder `C:\Program Files\blueCFD-Core-2017\ofuser-of5`

4. Place the downloaded package into this folder.

5. Go back to the terminal window and do the same steps you've done in the past for unpacking and compiling it.

----

**Possibility B**: If by any chance you've used the `git clone` command in the default home folder because the `user` command have failed in the past, please let me know, but if this was indeed the situation, then please run the following commands inside the terminal:

```
mkdir -p $FOAM_RUN
user
```

And it should work properly.

Best regards,
Bruno

**Comment by LeeLeong** [comment]
Hi Bruno,

Thanks for attend to my problem.  Indeed I am using the instruction from
https://github.com/blueCFD/Core/wiki/Building-cfMesh-with-blueCFD-Core#building-with-bluecfd-core-2017
. Therefore, I choose the solution of possibility B and it works.
Following your explaination, I found out I some how do not clone the cfMesh
inside 'user" home directory.  The only thing I can say is maybe I do not
login to "user" account.  But i remember I follow exactly the instruction.
Anyway, thank you very much for the help.  I attached here the log file of
the compilation progress.

Thanks ya.

Leong

**Comment by wyldckat** [comment]
I see... looks like I need to improve the description of the command `user`.

The command `user` is an _alias_ command. If you run the following command:

```
alias user
```

you will get the following output:

```
alias user='cd $WM_PROJECT_USER_DIR'
```

So when we run the command:

```
user
```

if will run the associated alias command:

```
cd $WM_PROJECT_USER_DIR
```

which will then go into the folder `/home/ofuser/blueCFD/ofuser-of5` on its own.

As a side note, the username `ofuser` is the default username blueCFD-Core always uses within the terminal, which means that everyone is sort-of logged into it by default. This was done to avoid having usernames with special characters or spaces in said name.

Many thanks for your feedback on this and I'll add this information to the FAQ (Frequently Asked Questions) page, along with links to it whenever we mention the command `user` when working on the next milestone.

**Comment by LeeLeong** [comment]
I appreciate your effort making this available to all of us.  Thank you
very much.
